Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs

```diff
@@ -200,15 +204,21 @@ private void ResolveLabel(ParserRuleContext context, string label)
                             ? member.DeclarationType.HasFlag(DeclarationType.Procedure)
                             : member.DeclarationType.HasFlag(DeclarationType.Function)));
             }
-
+            if (isRedimVariable)
+            {
+                // If the redim statement causes a variable to be created we have to add it manually to the state.      
+                // We rely on the fact that AddDeclaration removes/does not allow duplicates.
+                _state.AddDeclaration(boundExpression.ReferencedDeclaration);
```

Comment:
IIUC this makes the declaration be located at the *last* `ReDim` statement, but I believe the declaration should be at the *first* instance in the current scope. Not sure introducing RPS dependency is really needed here.